android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIBottomNavigationTabGroup.java

```diff
@@ -0,0 +1,260 @@
+
+		this.mBottomNavigationView = new BottomNavigationView(activity);
+		// Set the colorPrimary as backgroundColor by default if do not have the backgroundColor set.
+		if (proxy.hasPropertyAndNotNull(TiC.PROPERTY_TABS_BACKGROUND_COLOR)) {
+			this.mBottomNavigationView.setBackgroundColor(
+				TiColorHelper.parseColor(proxy.getProperty(TiC.PROPERTY_TABS_BACKGROUND_COLOR).toString()));
+		} else {
+			this.mBottomNavigationView.setBackgroundColor(this.colorPrimaryInt);
+		}
+		// Set the LayoutParams for the ViewPager.
+		TiCompositeLayout.LayoutParams params = new TiCompositeLayout.LayoutParams();
+		params.autoFillsWidth = true;
+		params.optionBottom = new TiDimension(mBottomNavigationHeightValue, TiDimension.TYPE_BOTTOM);
+
+		// Set the LayoutParams for the BottomNavigationView.
+		TiCompositeLayout.LayoutParams bottomNavigationParams = new TiCompositeLayout.LayoutParams();
+		bottomNavigationParams.autoFillsWidth = true;
+		bottomNavigationParams.optionBottom = new TiDimension(0, TiDimension.TYPE_BOTTOM);
+
+		// Add the views to their container.
+		((TiCompositeLayout) activity.getLayout()).addView(this.tabGroupViewPager, params);
+		((TiCompositeLayout) activity.getLayout()).addView(mBottomNavigationView, bottomNavigationParams);
+		// Set teh ViewPager as a native view.
+		setNativeView(this.tabGroupViewPager);
+	}
+
+	/**
+	 * Handle the removing of the controller from the UI layout when tab navigation is disabled.
+	 * @param disable
+	 */
+	@Override
+	public void disableTabNavigation(boolean disable)
+	{
+		super.disableTabNavigation(disable);
+		TiCompositeLayout.LayoutParams params = new TiCompositeLayout.LayoutParams();
+		params.autoFillsWidth = true;
+		if (disable) {
+			params.optionBottom = new TiDimension(0, TiDimension.TYPE_BOTTOM);
+			mBottomNavigationView.setVisibility(View.GONE);
+		} else {
+			params.optionBottom = new TiDimension(mBottomNavigationHeightValue, TiDimension.TYPE_BOTTOM);
+			mBottomNavigationView.setVisibility(View.VISIBLE);
+		}
+		this.tabGroupViewPager.setLayoutParams(params);
+	}
+
+	@Override
+	public void addTabItemInController(TabProxy tabProxy)
+	{
+		// Guard for the limit of tabs in the BottomNavigationView.
+		if (this.mMenuItemsArray.size() == 5) {
+			Log.e(TAG, "Trying to add more than five tabs in a TabGroup with TABS_STYLE_BOTTOM_NAVIGATION style.");
+			return;
+		}
+		// Create a new item with id representing its index in mMenuItemsArray.
+		MenuItem menuItem = this.mBottomNavigationView.getMenu().add(null);
+		// Set the click listener.
+		menuItem.setOnMenuItemClickListener(this);
+		// Set the title.
+		if (tabProxy.hasPropertyAndNotNull(TiC.PROPERTY_TITLE)) {
+			menuItem.setTitle(tabProxy.getProperty(TiC.PROPERTY_TITLE).toString());
+		}
+		// Set the icon.
+		if (tabProxy.hasPropertyAndNotNull(TiC.PROPERTY_ICON)) {
+			Drawable drawable = TiUIHelper.getResourceDrawable(tabProxy.getProperty(TiC.PROPERTY_ICON));
+			menuItem.setIcon(drawable);
```

Comment:
There is an issue here with loading image icons for the top tabs from APK "assets". The old version of `TabBar` used to scale the images loaded from "assets". If I remember right, the old code assumed asset images were "mdpi" and scaled them to the device's current density. This new code loads the images as-is, which means they end up being extra tiny on a xxxhdpi device, such as my Pixel 2 phone.

Our `BottomNavigationView` does not have this problem and scales "asset" images. Perhaps you should use the same code?

Reply:
You can test this with our "tab1.png" and "tab2.png" files that our Titanium Classic app template comes with.

Reply:
Updated.